pkg/syncer/util.go

```diff
@@ -707,13 +708,39 @@ func createCnsVolume(ctx context.Context, pv *v1.PersistentVolume,
 	} else {
 		log.Infof("vSphere CSI Driver has successfully marked volume: %q as the container volume.",
 			pv.Spec.CSI.VolumeHandle)
-
-		if isMultiVCenterFssEnabled && len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
+		if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
+			if isMultiVCenterFssEnabled && len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
+				// Create CNSVolumeInfo CR for the volume ID.
+				err = volumeInfoService.CreateVolumeInfo(ctx, pv.Spec.CSI.VolumeHandle, vcHost)
+				if err != nil {
+					log.Errorf("Failed to store volumeID %q for vCenter %q in CNSVolumeInfo CR. Error: %+v",
+						pv.Spec.CSI.VolumeHandle, vcHost, err)
+					return err
+				}
+			}
+		} else if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorWorkload &&
+			commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.PodVMOnStretchedSupervisor) {
+			k8sclient, err := k8s.NewClient(ctx)
+			if err != nil {
+				log.Errorf("Failed to initialize K8S client. Error: %+v", err)
+				return err
+			}
+			scList, err := k8sclient.StorageV1().StorageClasses().List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return logger.LogNewErrorf(log, "Failed to list Storageclasses from API server. Error: %+v", err)
+			}
+			var storagePolicyID string
+			for _, sc := range scList.Items {
```

Comment:
I did not understand this logic.. why are we picking the first SC we get in the SC list? Shouldn't we choose the SC related to the PV?

Reply:
Right, I initially had the code written assuming storageclass name is not available. 
Changed the logic to use StorageClass.Get()